src/weights.jl

```diff
@@ -35,7 +36,33 @@ end
     W(v, sum(v))
 end
 
-Base.getindex(wv::W, ::Colon) where {W <: AbstractWeights} = W(copy(wv.values), sum(wv))
+Base.getindex(wv::AbstractWeights, ::Colon) = copy(wv)
+
+Base.copy(wv::W) where {W <: AbstractWeights} =
+    weightstype(W)(copy(wv.values), sum(wv))
+
+@propagate_inbounds function Base.view(wv::W, inds...) where {S <: Real, W <: AbstractWeights{S}}
+    @boundscheck checkbounds(wv, inds...)
+    @inbounds v = invoke(view, Tuple{AbstractArray, Vararg{Any}}, wv, inds...)
+    # Sum is not actually used but compute the right type for clarity
+    weightstype(W)(v, zero(S))
+end
+
+# This method is implemented for backward compatibility
```

Comment:
I understand the reason is that this 0-dimensional view must not be `AbstractWeights` as weights must be exactly 1-dimensional.

Reply:
Right. This is just to avoid breaking the cases you spotted above (even though in practice they are unlikely to be used).